edweiss-app/model/lectures/lectureDoc.ts

```diff
@@ -49,8 +50,8 @@ namespace LectureDisplay {
 
 	export const Functions = FunctionFolder("lectures", {
 		addAudioTranscript: FunctionOf<{ courseId: string, lectureId: string, pageNumber: number, transcription: string; }, {}, 'invalid_arg' | 'error_firebase' | 'successfully_added'>("addAudioTranscript"),
-		createQuestion: FunctionOf<{ courseId: string, lectureId: string, question: string; }, { id: string; }, 'invalid_arg' | 'error_firebase' | 'empty_question'>("createQuestion"),
-		updateQuestion: FunctionOf<{ id: string, lectureId: string, likes: number; }, { id: string; }, 'invalid_id' | 'error_firebase' | 'empty_question'>("updateQuestion"),
+		createQuestion: FunctionOf<{ courseId: string, lectureId: string, question: string, username: string; }, { id: string; }, 'invalid_arg' | 'error_firebase' | 'empty_question'>("createQuestion"),
```

Comment:
Again, the username should not be an argument of the Cloud Function, as it one could easily usurp someone's identity.
In your cloud function, try the following :
```typescript
const user = getRequiredDocument(Collections.users, userId, USER_NOT_FOUND);
user.name
```

Reply:
Yes thank you, indeed if I use the useUser() that you mentioned this is no longer necessary